test/test_reductions.py

```diff
@@ -3090,6 +3090,30 @@ def test_histc_lowp(self, device, dtype):
             actual)
         self.assertEqual(actual.dtype, dtype)
 
+    @dtypes(torch.int8, torch.int, torch.long, torch.float, torch.double)
```

Comment:
Are torch.half, and torch.bfloat16 not supported by this op?

Reply:
@Skylion007 Seems not supported yet, running tests will get errors.

```bash
pytest test/test_reductions.py -k test_histc_min_max_corner_cases_cuda -vv
```

```python
    @onlyCUDA
    @dtypes(torch.int8, torch.int, torch.long, torch.bfloat16, torch.float16)
    def test_histc_min_max_corner_cases_cuda(self, device, dtype):
        actual = torch.histc(
            torch.tensor([1., 2, 1], dtype=dtype, device=device),
            bins=4, min=5, max=5)
        self.assertEqual(
            torch.tensor([2, 0, 0, 1], dtype=dtype, device=device),
            actual)
```


![image](https://github.com/user-attachments/assets/547efc28-fad0-49c4-b772-ba731b50885c)

I think one of them comes from here:
https://github.com/pytorch/pytorch/blob/e6e140c3d731a15fbdb1a8df93a288915c4c0e97/aten/src/ATen/native/cuda/SummaryOps.cu#L390

If it's better to support these types, I'm willing to make it after this. :D